pkg/reconciler/pipeline/controller.go

```diff
@@ -725,6 +725,15 @@ func (r *pipelineReconciler) updateDesiredState(ctx context.Context, pl *dfv1.Pi
 }
 
 func (r *pipelineReconciler) resumePipeline(ctx context.Context, pl *dfv1.Pipeline) (bool, error) {
+
+	// reset pause timestamp
+	if pl.GetAnnotations()["pauseTimestamp"] != "" {
+		delete(pl.GetAnnotations(), "pauseTimestamp")
+		if err := r.client.Update(ctx, pl.DeepCopy()); err != nil {
```

Comment:
Do not update the pipeline here, instead, return the pipeline object with modified data, and let the reconciler do it. https://github.com/numaproj/numaflow/blob/main/pkg/reconciler/pipeline/controller.go#L88

You also need to change the logic of `needsUpdate()`